src/dag/dag_block.hpp

```diff
@@ -9,15 +11,23 @@ using VdfSortition = vdf_sortition::VdfSortition;
 
 // Note: Need to sign first then sender() and hash() is available
 class DagBlock {
-  blk_hash_t pivot_;
-  level_t level_ = 0;
-  vec_blk_t tips_;
-  vec_trx_t trxs_;  // transactions
-  sig_t sig_;
-  blk_hash_t hash_;
-  uint64_t timestamp_ = 0;
-  vdf_sortition::VdfSortition vdf_;
-  mutable addr_t cached_sender_;  // block creater
+ public:
+  /**
+   * @brief list of Dag block RLP fields
+   *
+   * @note internal values cannot be changed and PIVOT must have value 0 as that is rlp fixed fields structure
+   */
+  enum RlpField : uint8_t {
```

Comment:
this enum is just an additional verbosity without added value. It was pretty obvious what field has what position in the encoding/decoding code

Reply:
well it was not so obvious and writing things like rlp[3], etc... in code is just terrible. In case you need to work with that rlp you have to chcek the structure, think about it find the index and use that. Now you can just use enum, which is much easier to read and use.

Reply:
I will replace it with a template-based encoding/decoding like in `enc_rlp` - this way it's clean and no need for enums

Reply:
cool, looking forwards for that